clang/test/CodeGenSYCL/sycl-device.cpp

```diff
@@ -3,6 +3,15 @@
 
 int bar(int b);
 
+class A {
+public:
+  // TODO: check for constructor/destructor as well
```

Comment:
One more `TODO`: check how it works for class templates and class member templates

Reply:
> One more TODO: check how it works for class templates and class member templates

Not sure that it's so hard to defer it to another patch.